pkg/controllers/managementagent/ingresshostgen/ingresshostgen_test.go

```diff
@@ -0,0 +1,229 @@
+package ingresshostgen
+
+import (
+	"testing"
+
+	rnetworkingv1 "github.com/rancher/rancher/pkg/generated/norman/networking.k8s.io/v1"
+	rnetworkingv1fakes "github.com/rancher/rancher/pkg/generated/norman/networking.k8s.io/v1/fakes"
+	"github.com/rancher/rancher/pkg/ingresswrapper"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	knetworkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakediscovery "k8s.io/client-go/discovery/fake"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+type IngressInterface struct {
+	ingressInterface rnetworkingv1.IngressInterface
+}
+
+func (i *IngressInterface) Ingresses(namespace string) rnetworkingv1.IngressInterface {
```

Comment:
Could this use [IngressGetterMock](https://github.com/rancher/rancher/blob/release/v2.8/pkg/generated/norman/networking.k8s.io/v1/fakes/zz_generated_ingress_mock.go#L1742-L1754) instead of adding new mock methods here?

Reply:
I could have the following instead:

```go
type IngressInterface struct {
	*rnetworkingv1fakes.IngressesGetterMock
	*rnetworkingv1fakes.NetworkPoliciesGetterMock
}

...

	mockedIngressesGetter := &rnetworkingv1fakes.IngressesGetterMock{
		IngressesFunc: func(namespace string) rnetworkingv1.IngressInterface {
			return &rnetworkingv1fakes.IngressInterfaceMock{
				UpdateFunc: func(obj *rnetworkingv1.Ingress) (*rnetworkingv1.Ingress, error) {
					return obj, nil
				},
			}
		},
	}
	ingressCompat := ingresswrapper.NewCompatInterface(&IngressInterface{
		IngressesGetterMock:       mockedIngressesGetter,
		NetworkPoliciesGetterMock: &rnetworkingv1fakes.NetworkPoliciesGetterMock{},
	}, nil, clientset)
	gen := &IngressHostGen{
		ingress: ingressCompat,
	}


```

Does that look more like something we'd want? I at least need the `IngressInterface` struct for both interfaces that are needed, so I kept it, and instead I embed the different mocks into that struct directly.

Reply:
You're right, my mistake, the `IngressGetterMock` isn't sufficient here and there's no mock for the `Interface` struct which is what we need.

I like the alternative you suggested. The only thing is that instead of `type IngressInterface struct`, I think it would be better as `type NetworkingInterface struct` since the mock isn't just for `Ingresses` but also other `networking.k8s.io` APIs, including `NetworkPolicies`.

Reply:
Good idea.  I rebased and made the changes above + renamed to the better name `NetworkingInterface`